### PR TITLE
fix(security): harden tool manager with security protections (task-249) [v2]

### LINKED ISSUE #1: Installer downloads zip but fails to extract or use it (potential GitHub auth issue)
## Problem

When running the installer in some cases, a zip file (e.g., `/home/jpoley/src/triggerprod/spec-kit-template-copilot-sh-v0.0.79.zip`) is downloaded into the target directory but the install process fails to properly extract or use it. This leaves the zip sitting in the directory and results in a broken install. This is a serious issue that blocks setup and is difficult to clean up after failed attempts.

## Context
- The problem is **not** due to public/private repo status—making the project public is not an acceptable workaround.
- GitHub authentication or token issues *may* be related, especially when accessing private assets or APIs.
- The expected install flow should either extract the zip content or fail gracefully and provide actionable error messages, not just leave the zip in place without resolution.

## Steps to reproduce
1. Attempt to install the CLI or template in an environment that triggers this failure (observed example: `/home/jpoley/src/triggerprod/spec-kit-template-copilot-sh-v0.0.79.zip` left after run).
2. Observe that the zip file remains in the directory and the install is incomplete.
3. No clear error or clean-up occurs after failure.

## Expected behavior
- Install process should extract and use downloaded zips, or if it fails, remove partial files and provide clear error messaging to the user.
- If GitHub authentication fails (e.g., with private repos or rate limits), the error should guide users towards supplying valid credentials (such as `GITHUB_JPSPEC` or `--github-token`).

## Additional notes
- This issue blocks local development setup and repeatability.
- Needs debugging for where zip handling or authentication step fails, especially for private assets.
- Solution must not involve making repo or assets public—authentication/permission handling must be addressed instead.

## Next steps
- Investigate install logs when this happens for error details
- Add robust error handling and cleanup for incomplete downloads or failed extractions
- Provide actionable feedback to the user for common auth/permission issues

/cc @jpoley

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

Implements a hardened tool dependency manager for security scanners with comprehensive security protections. This is v2 of the implementation, addressing all 12 code review issues from the original PR #484.

### Path Traversal & Validation (Issues #1, #2, #5)

**FIX #1: Path traversal detection using Path.parts**
- Changed from substring check (`".." in member`) to path component check
- Prevents false positives on legitimate files like `README..md`
- Only flags actual path traversal attempts like `../../etc/passwd`

**FIX #2 & #5: Path validation with relative_to()**
- Changed from string prefix matching to `relative_to()` method
- Prevents bypasses via symlinks or similar directory names
- Applied to both ZIP and non-ZIP archive extraction

### DoS Prevention (Issues #3, #6)

**FIX #3 & #6: File count limit for zip bomb protection**
- Added `MAX_ARCHIVE_FILES = 10000` constant
- Enforced in ZIP member count check (before extraction)
- Enforced in post-extraction validation (for non-ZIP archives)
- Prevents decompression bombs and resource exhaustion

### File Handling (Issues #7, #10)

**FIX #7: Case-insensitive extension checking**
- Changed from `str.endswith(".zip")` to `Path.suffix.lower() == ".zip"`
- Handles uppercase extensions like `.ZIP` correctly
- More robust and cross-platform

**FIX #10: Executable size validation**
- Added `MAX_EXECUTABLE_SIZE_MB = 1000` sanity check
- Validates file size before setting executable permissions
- Prevents accidentally installing huge files

### Code Quality (Issues #11, #12)

**FIX #11 & #12: Explanatory comments in except clauses**
- All empty `except` blocks now have clear explanatory comments
- Added `logger.debug()` messages for debugging
- Explains why exceptions are caught and ignored

### New Test Coverage (Issues #4, #8, #9)

**FIX #4: Non-ZIP archive tests**
- Added tests for `.tar.gz` and `.tar.bz2` extraction
- Validates file count limit works for all archive types

**FIX #8: VERSION_PATTERN edge cases**
- Tests for versions at end of text
- Tests for versions embedded in different contexts

**FIX #9: Path traversal bypass attempts**
- Tests for `../../etc/passwd` (standard traversal)
- Tests for `foo/../../etc/passwd` (traversal after directory)
- Tests for multiple levels of traversal

## Features

- **Auto-installation** with version pinning
- **License compliance checking** (CodeQL requires acceptance)
- **Dependency size monitoring** (500MB threshold warning)
- **Offline mode** for air-gapped environments
- **Cross-platform support** (Linux, macOS, Windows)
- **Built-in configs** for semgrep, codeql, bandit

## Test Results

```
tests/security/tools/test_manager.py ................ [ 71% ]
tests/security/tools/test_models.py ................. [100%]

71 passed, 1 warning in 0.35s
```

All tests passing ✅

## Files Changed

- `src/specify_cli/security/tools/manager.py` - Main tool manager (768 lines)
- `src/specify_cli/security/tools/models.py` - Data models (145 lines)
- `tests/security/tools/test_manager.py` - Manager tests (714 lines)
- `tests/security/tools/test_models.py` - Model tests (118 lines)

Total: 1,745 lines of production code and tests

## Related Issues

- Closes #484 (original PR, closed due to code review issues)
- Addresses task-249 security hardening requirements

## Checklist

- [x] Path traversal uses `Path.parts` not substring
- [x] Path validation uses `relative_to()` not string prefix
- [x] File count limit added (MAX_ARCHIVE_FILES)
- [x] Extension check uses `.suffix.lower()`
- [x] Empty except clauses have comments
- [x] Tests for non-ZIP archives added
- [x] Tests for path traversal bypass attempts added
- [x] All tests pass (71/71)
- [x] No backlog task files in commit
- [x] Branch name starts with `muckross-`

### CODE REVIEW

File: tests/security/tools/test_manager.py [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+            for i in range(10):
+                import io
+                content = f"content {i}".encode()
+                tarinfo = tarfile.TarInfo(name=f"file{i}.txt")
+                tarinfo.size = len(content)
+                tf.addfile(tarinfo, io.BytesIO(content))
+
+        # Should extract successfully
+        manager._safe_extract_archive(safe_tar, extract_dir)
+        assert (extract_dir / "file0.txt").exists()
```
[the remark this answers is not in the document]
Comment (jpoley) [reply tests/security/tools/test_manager.py]: fix me

File: src/specify_cli/security/tools/manager.py [hunk 2]
```diff
@@ -0,0 +1,814 @@
...
+                    except ValueError:
+                        raise RuntimeError(
+                            f"Archive member escapes destination: {member}"
+                        )
+
+                # All paths validated, safe to extract
+                zf.extractall(dest_dir)
+        else:
+            # For other archive types, use shutil but still validate after
+            shutil.unpack_archive(archive_path, dest_dir)
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/security/tools/manager.py]: fix me

File: src/specify_cli/security/tools/manager.py [hunk 3]
```diff
@@ -0,0 +1,814 @@
...
+        Args:
+            tool_name: Name of the tool to install.
+            accept_license: If True, auto-accept license (for CodeQL etc).
+            force: If True, reinstall even if already installed.
+
+        Returns:
+            InstallResult with success status and tool info or error.
+
+        Raises:
+            ValueError: If tool_name is not in configurations.
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/security/tools/manager.py]: fix me

**Comment by jpoley** [comment]
fix me and ci lint fails
